code/modules/surgery/organs/heart.dm

```diff
@@ -146,5 +146,104 @@
 	icon_state = "heart-c-on"
 	icon_base = "heart-c"
 	dead_icon = "heart-c-off"
-	origin_tech = "biotech=5"
-	status = ORGAN_ROBOT
\ No newline at end of file
+	status = ORGAN_ROBOT
+
+/obj/item/organ/internal/heart/cybernetic/upgraded
+	name = "upgraded cybernetic heart"
+	desc = "A more advanced version of a cybernetic heart. Grants the user additional stamina and heart stability, but the electronics are vulnerable to shock."
+	icon_state = "heart-c-u-on"
+	icon_base = "heart-c-u"
+	dead_icon = "heart-c-u-off"
+	var/emagged = FALSE
+	var/attempted_restart = FALSE
+
+/obj/item/organ/internal/heart/cybernetic/upgraded/on_life()
+	if(!ishuman(owner))
+		return
+	if(!(status & ORGAN_DEAD) && !attempted_restart && !beating)
+		to_chat(owner, "<span class='warning'>Your [name] detects a cardiac event and attempts to return to its normal rhythm!</span>")
```

Comment:
Now this is just me going for aesthtic..but it should be phrased as a little robot voice in your head informing you of heart errors (not all of them useful like true errors)